src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs

```diff
@@ -189,6 +190,57 @@ public async Task StoringDepositsWhenWalletBalanceSufficientSucceedsWithDetermin
             }
         }
 
+
+        /// <summary>
+        /// Recording deposits when the wallet UTXOs are sufficient succeeds with deterministic transactions.
+        /// </summary>
+        [Fact]
+        public async void StoringDepositsWhenTargetIsContractFailsWithRejectedTransferAsync()
+        {
+            var dataFolder = new DataFolder(TestBase.CreateTestDir(this));
+
+            this.Init(dataFolder);
+            this.AddFunding();
+            this.AppendBlocks(WithdrawalTransactionBuilder.MinConfirmations);
+
+            MultiSigAddress multiSigAddress = this.wallet.MultiSigAddress;
+
+            using (ICrossChainTransferStore crossChainTransferStore = this.CreateStore())
+            {
+                crossChainTransferStore.Initialize();
+                crossChainTransferStore.Start();
+
+                TestBase.WaitLoopMessage(() => (this.ChainIndexer.Tip.Height == crossChainTransferStore.TipHashAndHeight.Height, $"ChainIndexer.Height:{this.ChainIndexer.Tip.Height} Store.TipHashHeight:{crossChainTransferStore.TipHashAndHeight.Height}"));
```

Comment:
Is the WaitLoop necessary here? The chain indexer and store seem to be initialized synchronously.

Reply:
The store is initialized to the wallet tip. I guess this would be confirming that the wallet and store are now both up to date with the chain indexer. It may not be strictly required but it does no harm having this.

Reply:
Right, and the store `Initialize` method seems to be synchronous so it will always be at the wallet tip at this point. I think it's not necessary to leave it in, but if it does stay there's no reason to use `WaitLoop` as opposed to an `Assert.Equal`.

Reply:
The change is out of scope for this PR as changing this now will affect multiple test cases.